test/e2e/case1_framework_deployment_test.go

```diff
@@ -166,6 +166,11 @@ var _ = Describe("Test framework deployment", func() {
 
 			checkContainersAndAvailability(cluster, i)
 
+			deploy = GetWithTimeout(
+				cluster.clusterClient, gvrDeployment, case1DeploymentName, addonNamespace, true, 30,
+			)
+			Expect(deploy).NotTo(BeNil())
+
```

Comment:
This is the same check as the first one before `checkContainersAndAvailability`. Should the first one be removed or could that cause additional flakiness?

Reply:
I decided to leave the original one in because `checkContainersAndAvailability` only checks the deployment if `startupProbeInCluster` returns true.

In the case of `startupProbeInCluster` returning false, the pods are directly checked with a 30 second timeout. So if I removed the first `GetWithTimeout` for deploy, then that case would timeout in 30 seconds instead of the 60 seconds + 30 seconds.